### ISSUE TITLE
Inductor doesn't fuse outer dimension softmax into a single kernel.

### ISSUE THREAD

**Issue by Chillee** [pytorch/torchdynamo#1543]
http://ix.io/4cog

We generate two kernels:

![image](https://user-images.githubusercontent.com/6355099/194673608-c62b6f4e-3be9-4872-a7d8-0ae2b01bc44e.png)

The issue seems to lie in the memory dependencies not matching up.

cc: @jansel @ngimel
